### DISCUSSION TITLE
osu!editor should be able to align a slider tail to a bpm change.

### DISCUSSION THREAD

**Discussion by Lunarien** [ppy/osu#25040] [Ideas]
I will make references to the old osu! client, the non lazer version throughout this post, but i don't think it will matter a lot, because the same problem+feature request seems applicable to lazer too.

https://github.com/ppy/osu/assets/73956350/347e63a5-3dd3-4c90-80a4-bd1898d3f858

In this video clip, you can hear the song having a sample that is kind of irregular in relation to the rest of the song. the bpm starts to be regular again from the endpoint of the sample.
The idea/suggestion is that for if I wanted to make a slider follow that sound, I would have to snap the slider tail to where the BPM change occurs. This is not possible as of now (and/or as far as i know), and I think it should be made possible to do that using the editor.

**Comment by Lunarien** [comment]
The only way one could accomplish this is through editing the file of the beatmap (a .osu file, if I remember correctly). This is a clumsy process though, and can take up a bunch of your time, either through revisions of the same beatmap, or through a song having multiple of these parts.
Also, I found an old but related post on the osu! forums that has a bit more to say about this: https://osu.ppy.sh/community/forums/topics/346077

**Reply by peppy** [reply]
We definitely want to add better support for BPM ramps and timing changes in general. I can't guarantee the specific scenario you mention will be supported, but in theory it should be feasible.
